### PR TITLE
Add TestCase that simplify assertion on http form

### PR DESCRIPTION
**Pull request by jderusse**

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -10,6 +10,7 @@ parameters:
         - src/*/Tests/*
         - src/**/Tests/*
         - src/CodeGenerator/Generator/PhpGenerator/ClassFactory.php
+        - src/Core/Test/TestCase.php
```
Comment (jderusse) [inline phpstan.neon.dist]: this class extends TestCase, and we don't requires phpunit

Comment (jderusse) [reply phpstan.neon.dist]: replace by a trait would fix that issue

File: src/CloudFormation/Tests/Unit/Input/DescribeStackEventsInputTest.php [hunk 2]
```diff
@@ -3,7 +3,7 @@
 namespace AsyncAws\CloudFormation\Tests\Unit\Input;
 
 use AsyncAws\CloudFormation\Input\DescribeStackEventsInput;
-use PHPUnit\Framework\TestCase;
+use AsyncAws\Core\Test\TestCase;
```
Comment (jderusse) [inline src/CloudFormation/Tests/Unit/Input/DescribeStackEventsInputTest.php]: not compatible with core 0.2 . provide a migration path.

Comment (Nyholm) [reply src/CloudFormation/Tests/Unit/Input/DescribeStackEventsInputTest.php]: Core 0.2 is not released yet. So this is fine

**Comment by Nyholm** [comment]
I think it is a.good idea to add a test case. I would be happy to merge after to make PHPStan happy. 

It is okey that Lowest Deps fails. The packages that used the new TestCase required core 0.2. 

If you really dont want to merge this with a red CI, you could split it up to two PRs. First PR where you only make changes to the core package. Once that is merged, you make the second PR with the rest of the changes.

**Pull request merged**